### ISSUE TITLE
Pagination: accessible "go to page" links

### ISSUE THREAD

**Issue by ygatesoupe** [twbs/bootstrap#19158]
Page links should mention to screen reader their function and which page they address, using aria-label attribute as following :

` <li class="page-item"><a class="page-link" href="#" aria-label="Go to page 1">1</a></li>`

Wrapping tag `<nav>` should define a navigation role :

`<nav role="navigation">`

**Comment by patrickhlauke** [comment]
Partially agree with the first point - making links more unambiguous would definitely be good. However, `<nav>` should really only be used for major/primary navigation (i.e. not every set of navigational links should be wrapped in a `<nav>`) - this will be dependent on how the pagination is actually used, so shouldn't be mandated/added indiscriminately to the example in the docs.

**Comment by cvrebert** [comment]
Would something like adding a `<p class="sr-only">Go to page…</p>` before the list of pages be an option? Individual `aria-label`s on each link sounds a tad verbose markup-wise.

**Comment by patrickhlauke** [comment]
That would work under the assumption that an SR user was going through the page linearly, so first reading that bit of text, then coming across the links. I'll take a straw-poll of what some colleagues think (as this is where we get more into "opinion" versus hard and fast rules)

**Comment by patrickhlauke** [comment]
rough consensus from colleagues seems to be that yes, adding individual `aria-label`s to each link may be overkill, but that it is definitely best to provide some form of explicit grouping, which can then provide a label, and that `<nav>` may indeed not always be the most appropriate, but won't do much harm if used. in short, the recommendation is something along the lines of
```
<nav aria-label="Page navigation">
  <ul ...>
    <li ...><a href="..." ...>1</a></li>
    ...
  </ul>
</nav>
```

**Comment by patrickhlauke** [comment]
Closed via #19704 / #19705
